### ISSUE TITLE
Bore lens can not be used to break sugar cane

### ISSUE THREAD

**Issue by DasGerippe** [Vazkii/Botania#3506]
# Version Information

Forge version:  1.16.3-forge-34.1.29
Botania version:  1.16.3-409

# Further Information

In previous versions it was possible to use a mana spreader with a bore lens to break sugar cane. This isn't possible in 1.16.3 anymore.

**Comment by delucain** [comment]
I don't know if it's related, but the Force Lens doesn't break them either. It seems like it should, since Force Lenses replicate the behavior of pistons.

**Comment by rehashedsalt** [comment]
I was able to replicate this on the latest Curseforge-published* version of Botania on Fabric as well:

* Minecraft 1.18.1
* Fabric Loader 0.12.12
* Fabric API 0.45.0+1.18
* Botania 1.18.1-427-FABRIC
* Patchouli 1.18.1-63-FABRIC
* Trinkets 3.1.0

Would be great to have this hotfixed into the 1.16 branch, too.

**Comment by williewillus** [comment]
we're not sure how this is going to go. Basically, mana bursts used to have weird logic and code to collide with blocks with no real collision box. Not keen to reintroduce that hackery as it's a maintenance burden and no other projectile in vanilla behaves the same way. The alternative for now is breaking the sugar canes with pistons.

**Comment by rehashedsalt** [comment]
The previous behavior definitely had its advantages, being overall cheaper to build, having fewer moving parts, and pairing well with the density that planting on Vivid Grass allowed. It's also less frame-killing to look at than a lot of vanilla farms.

If a good solution can't be found for mana bursts, I'd say adding sugar cane to the Drum of the Wild might not be the worst idea, even (or especially) if it breaks bases.

**Comment by RubyRezal** [comment]
+1 for adding suggar cane to the Drum of the Wild's list of breakables.  Suggar Cane is a plant after all and technically a crop.

**Comment by Cicopath** [comment]
I also agree with having the Drum of the Wild break sugar cane; it's a good alternative.

It's unfortunate that the sugar cane challenge cannot currently be completed as intended.

**Comment by hakanai** [comment]
> It's unfortunate that the sugar cane challenge cannot currently be completed as intended.

The challenge text only says bonus points for not using pistons. It doesn't say you can't use Force Lens. Use Force Lens to push a block into the sugar cane and then another force lens to push it back.

Still creates quite a mess for my setup because I was planning to use 3 in a row to harvest kelp as well.

**Comment by Joblerone** [comment]
> > It's unfortunate that the sugar cane challenge cannot currently be completed as intended.
> 
> The challenge text only says bonus points for not using pistons. It doesn't say you can't use Force Lens. Use Force Lens to push a block into the sugar cane and then another force lens to push it back.
> 
> Still creates quite a mess for my setup because I was planning to use 3 in a row to harvest kelp as well.

having attempted this challenge on this version and having thrown many hours into testing this, a sugarcane is treated as immovable by force lens, so all pushed blocks will collide with the sugarcane, failing to break it, then stop up against the side of it. attempting to push it again does not work - the sugarcane may as well be obsidian.

having said that, i agree, the challenge text ought to be updated to reflect the inability to do this pistonless with just botania, because i'm not getting those hours of my life back

**Comment by TheRealWormbo** [comment]
Note that there seems to be a bug that prevents force relays from doing their job in certain specific situations: #4593

**Comment by Joblerone** [comment]
> Note that there seems to be a bug that prevents force relays from doing their job in certain specific situations: #4593

this is not quite that - i pushed just an ordinary grass block directly to test this, just as a sanity check, and both the grass block and any sticky blocks i attempted to push into it were stopped

**Comment by TheRealWormbo** [comment]
From my tests, the force lens has no issues pushing solid blocks through sugarcane. The only thing that breaks is moving a force relay directly with a force lens or other force relay. Any indirect movement, or pushing things without fore relays should work.
